src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpTypeStyleDiagnosticAnalyzerBase.cs

```diff
@@ -11,12 +11,84 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.Diagnostics.TypeStyle.CSharpTypeStyleDiagnosticAnalyzerBase;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.TypeStyle
 {
+    internal abstract partial class CSharpTypeStyleHelper
+    {
+        internal abstract bool IsStylePreferred(SemanticModel semanticModel, OptionSet optionSet, State state, CancellationToken cancellationToken);
+        internal abstract bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken);
+        protected abstract bool AssignmentSupportsStylePreference(SyntaxToken identifier, TypeSyntax typeName, ExpressionSyntax initializer, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken);
+
+        internal TypeSyntax CanOfferFix(SyntaxNode declarationStatement, SemanticModel semanticModel, CancellationToken cancellationToken)
```

Comment:
we tend to prefer 'public' methods in internal types.  but this is fine if you prefer this.

Reply:
I'll leave as-if if that's ok.

---
In reply to: [172284161](https://github.com/dotnet/roslyn/pull/25208#discussion_r172284161) [](ancestors = 172284161)